lightning/src/ln/channelmanager.rs

```diff
@@ -3321,6 +3383,98 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
+	/// passed directly to [`claim_funds`].
+	///
+	/// See [`create_inbound_payment_for_hash`] for detailed documentation on behavior and requirements.
+	///
+	/// [`claim_funds`]: Self::claim_funds
+	/// [`PaymentReceived`]: events::Event::PaymentReceived
+	/// [`PaymentReceived::payment_preimage`]: events::Event::PaymentReceived::payment_preimage
+	/// [`create_inbound_payment_for_hash`]: Self::create_inbound_payment_for_hash
+	pub fn create_inbound_payment(&self, min_value_msat: Option<u64>, invoice_expiry_delta_secs: u32, user_payment_id: u64) -> (PaymentHash, PaymentSecret) {
+		let payment_preimage = PaymentPreimage(self.keys_manager.get_secure_random_bytes());
+		let payment_hash = PaymentHash(Sha256::hash(&payment_preimage.0).into_inner());
+
+		(payment_hash,
+			self.set_payment_hash_secret_map(payment_hash, Some(payment_preimage), min_value_msat, invoice_expiry_delta_secs, user_payment_id)
+				.expect("RNG Generated Duplicate PaymentHash"))
+	}
+
+	/// Gets a [`PaymentSecret`] for a given [`PaymentHash`], for which the payment preimage is
+	/// stored external to LDK.
+	///
+	/// A [`PaymentReceived`] event will only be generated if the [`PaymentSecret`] matches a
+	/// payment secret fetched via this method or [`create_inbound_payment`], and which is at least
+	/// the `min_value_msat` provided here, if one is provided.
+	///
+	/// The [`PaymentHash`] (and corresponding [`PaymentPreimage`]) must be globally unique. This
+	/// method may return an Err if another payment with the same payment_hash is still pending.
+	///
+	/// `user_payment_id` will be provided back in [`PaymentReceived::user_payment_id`] events to
+	/// allow tracking of which events correspond with which calls to this and
+	/// [`create_inbound_payment`]. `user_payment_id` has no meaning inside of LDK, it is simply
+	/// copied to events and otherwise ignored. It may be used to correlate PaymentReceived events
+	/// with invoice metadata stored elsewhere.
+	///
+	/// `min_value_msat` should be set if the invoice being generated contains a value. Any payment
+	/// received for the returned [`PaymentHash`] will be required to be at least `min_value_msat`
+	/// before a [`PaymentReceived`] event will be generated, ensuring that we do not provide the
+	/// sender "proof-of-payment" unless they have paid the required amount.
+	///
+	/// `invoice_expiry_delta_secs` describes the number of seconds that the invoice is valid for
+	/// in excess of the current time. This should roughly match the expiry time set in the invoice.
+	/// Note that we use block header time to time-out pending inbound payments (with some margin
+	/// to compensate for the inaccuracy of block header timestamps). Thus, in practice we will
+	/// accept a payment and generate a [`PaymentReceived`] event for some time after the expiry.
+	/// If you need exact expiry semantics, you should enforce them upon receipt of
```

Comment:
Maybe we could give a hint for how users should compute this number (i.e., should it be at least the highest `cltv_expiry_delta` of their route hints, if that makes sense?)

Reply:
Hmm, good point. I'm not sure what it should be - the limit is unrelated to the cltv_expiry, which is relative to the block height at the time the invoice is paid/received, but the timeout is relative to when the invoice is generated. Its really up to the user for how long they want the invoice to still be valid. Maybe we can better capture that requirement in the docs (or mention the default of 2 hours) - do you have any suggestions?

Reply:
Ah, I see. I think it may have just been my misunderstanding, hmm. I think mentioning the default would be helpful.

Reply:
Pushed a fixup commit, let me know if its good and I'll squash + merge.

Reply:
LGTM